node/engine/chainservice/eth_chainservice.go

```diff
@@ -387,6 +420,7 @@ func (ecs *EthChainService) subscribeForLogs() (chan error, ethereum.Subscriptio
 	// Subscribe to Adjudicator events
 	eventQuery := ethereum.FilterQuery{
 		Addresses: []common.Address{ecs.naAddress},
+		Topics:    [][]common.Hash{topicsToWatch},
```

Comment:
The addition of `Topics` creates a more targeted event subscription so we don't have to filter out unwanted topics later.